internal/repo/datasource/sql_test.go

```diff
@@ -0,0 +1,321 @@
+				return dB
+			},
+			validator: func(rows []model.Transaction, count int, err error) {
+				temp := []model.Transaction{{
+					TransactionId: "0000-1111-2222-3333",
+					AccountNumber: 1,
+					UserId:        "4444-1111-2222-3333",
+					Amount:        1000,
+					TransferTo:    1234567890,
+					CreatedAt:     time.Date(2023, time.December, 1, 1, 1, 1, 0, time.UTC),
+					UpdatedAt:     time.Date(2023, time.December, 1, 1, 1, 1, 0, time.UTC),
+					Status:        "approved",
+					Type:          "debit",
+					Comment:       "no comments",
+				}}
+				if err != nil {
+					t.Errorf("Want: %v, Got: %v", nil, err)
+					return
+				}
+				if count != 1 {
+					t.Errorf("Want: %v, Got: %v", 3, count)
+					return
+				}
+				if !reflect.DeepEqual(rows, temp) {
+					t.Errorf("Want: %v, Got: %v", temp, rows)
+					return
+				}
+			},
+		},
+		{
+			name:   "FAILURE::Get:: query error",
+			filter: map[string]interface{}{"userid": "1234"},
+			setupFunc: func() sqlDs {
+				db, mock, err := sqlmock.New()
+				if err != nil {
+					t.Fail()
+				}
+				dB := sqlDs{
+					sqlSvc: db,
+					table:  "newTemp",
+				}
+				mock.ExpectQuery(regexp.QuoteMeta("SELECT COUNT(`transaction_id`) FROM newTemp WHERE userid = '1234'")).WillReturnError(nil).WillReturnRows(sqlmock.NewRows([]string{"transaction_id"}).AddRow("1").AddRow("2").AddRow("3"))
+				mock.ExpectQuery("SELECT transaction_id, account_number, user_id, amount, transfer_to, created_at, updated_at, status, type, comment FROM newTemp WHERE userid = '1234' ORDER BY created_at LIMIT 1 OFFSET 2 ;").WillReturnError(errors.New("Unknown column"))
+				return dB
+			},
+			validator: func(rows []model.Transaction, count int, err error) {
+				if !strings.Contains(err.Error(), "Unknown column") {
+					t.Errorf("Want: %v, Got: %v", "Unknown column", err)
+				}
+			},
+		},
+		{
+			name:   "FAILURE::Get:: query error::2",
+			filter: map[string]interface{}{"userid": "1234"},
+			setupFunc: func() sqlDs {
+				db, mock, err := sqlmock.New()
+				if err != nil {
+					t.Fail()
+				}
+				dB := sqlDs{
+					sqlSvc: db,
+					table:  "newTemp",
+				}
+				mock.ExpectQuery(regexp.QuoteMeta("SELECT COUNT(`transaction_id`) FROM newTemp WHERE userid = '1234'")).WillReturnError(errors.New("query error")).WillReturnRows(sqlmock.NewRows([]string{"transaction_id"}).AddRow("1").AddRow("2").AddRow("3"))
+				mock.ExpectQuery("SELECT transaction_id, account_number, user_id, amount, transfer_to, created_at, updated_at, status, type, comment FROM newTemp WHERE user_id = '1234' ORDER BY created_at LIMIT 1 OFFSET 2 ;").WillReturnError(errors.New("Unknown column"))
```

Comment:
Looks like the mock is to return failure which is same as the above test case. There will be 1 test case where Query return failure for the count sql query.

Reply:
done

Reply:
Is the 2nd mock getting hit? Registering the 2nd mock to fail the get transaction query should not be hit as the error should be returned from when count query fails.
If the mock is registered and not hit, it should have given some sort of error. This suggests that the way of using the sql mock is incorrect. Check the docs and confirm on the usage.

Reply:
Is the 2nd mock getting hit? Registering the 2nd mock to fail the get transaction query should not be hit as the error should be returned from when count query fails.
If the mock is registered and not hit, it should have given some sort of error. This suggests that the way of using the sql mock is incorrect. Check the docs and confirm on the usage.

Reply:
done i have moved the expectation of second mock as t is not getting hit

Reply:
Validation on whether the mocks are actually getting hit as per the expectation set is still pending.
<img width="1038" alt="image" src="https://user-images.githubusercontent.com/77770177/218057108-f2606cbf-4f38-4a59-8c39-bbe86bfc6881.png">

Reply:
even if there is no characters like back tic or line break still the error is same

Reply:
done